server/src/engine.rs

```diff
@@ -1,31 +1,100 @@
-//! Implementation of pricing logics based on the ride type.
+//! Implementation of pricing logics based on the service type.
 
 use log::{debug, info};
+use snafu::prelude::Snafu;
 
-use crate::pricing_grpc::{pricing_request::ServiceType, PricingRequest};
+use crate::pricing_grpc::{pricing_request::ServiceType, PricingRequest, PricingRequests};
 
-/// Get pricing for a given query
-pub fn get_pricing(query: PricingRequest) -> f32 {
-    info!("Getting pricing for service type: {:?}", query.service_type);
+/// Errors that can occur when getting pricing.
+#[derive(Snafu, Copy, Clone, Debug, PartialEq)]
+pub enum PricingError {
+    /// Pricing requests contain multiple service types.
+    #[snafu(display("All pricing requests must have the same service type"))]
+    MultipleServiceTypes,
 
-    match ServiceType::from_i32(query.service_type) {
+    /// Request contains negative weight.
+    #[snafu(display("Request contains negative weight"))]
+    NegativeWeight,
+
+    /// Request contains negative distance.
+    #[snafu(display("Request contains negative distance"))]
+    NegativeDistance,
+
+    /// No pricing requests were provided.
+    #[snafu(display("No pricing requests were provided"))]
+    NoRequests,
+
+    /// Unknown service type error.
+    #[snafu(display("Unknown service type; cannot parse service type"))]
+    UnknownServiceType,
+}
+
+/// Get pricing given a [`PricingRequests`], which contains an array of
+/// [`PricingRequest`]s.
+///
+/// The array of [`PricingRequest`]s is used to allow pricing for
+/// multiple legs of a trip to be calculated.
+///
+/// # Notes
+/// The `service_type` field of all [`PricingRequest`]s in the array
+/// must be the same. It means the pricing service currently does not
+/// allow for a trip that involves different types of services.
+///
+/// This may change in the future as per business requirements.
+///
+/// # Returns
+/// If no errors occur, returns a vector of prices in dollars,
+/// corresponding to the order of the input requests.
+pub fn get_pricing(query: PricingRequests) -> Result<Vec<f32>, PricingError> {
+    let requests = query.requests;
+    info!("Getting pricing for {:?} requests", requests.len());
```

Comment:
consider making this a debug call with (get_pricing) prepended

Reply:
[17d9bf3](https://github.com/Arrow-air/svc-pricing/pull/19/commits/17d9bf3b229051e65e10107c1e7d5756c5de34e0)